src/standalone/Snow/snow_parameterizations.jl

```diff
@@ -372,3 +376,178 @@ function energy_from_T_and_swe(S::FT, T::FT, parameters) where {FT}
+    A = exp(
+        density.c3 *
+        c5 *
+        Δt_t *
+        exp(density.c4 * Tsnow - cx * (ρ - density.ρ_d)),
+    )
+    factor = factor * A
+
+    ρ_new = minimum([ρ * factor, FT(0.45)])
+    ρ_new = (ρ_new < FT(0.05)) ? ρ : ρ_new
+    return ρ_new
+end
+
+"""
+    newsnow_density(air_temp::FT)::FT where {FT}
+Estimates the density of newly fallen snow as a function of air temperature, according to the Snow17 model.
+Used in computing the time derivative of snow depth under the Anderson1976 density parameterization.
+"""
+function newsnow_density(air_temp::FT)::FT where {FT}
+    if air_temp < -15
+        return FT(0.05)
+    else
+        return FT(0.05) + FT(0.0017) * FT((air_temp + FT(15.0))^1.5)
+    end
+end
+
+"""
+    newsnow_temp(air_temp::FT)::FT where {FT}
+Estimates the temperature of newly fallen snow as a function of air temperature, according to the Snow17 model.
+Used in computing the time derivative of snow depth under the Anderson1976 density parameterization.
+"""
+function newsnow_temp(air_temp::FT)::FT where {FT}
+    return (air_temp > 0) ? FT(0) : air_temp
+end
+
+"""
+    dzdt(density::Anderson1976{FT}, model::SnowModel{FT}, Y, p, t) where {FT}
+Returns the change in snow depth (rate) given the current model state and the `Anderson1976`
+density paramterization, which estimates contributions from new snowfall, as well as compaction
+effects from the existing and newly fallen snow.
+"""
+function dzdt(density::Anderson1976{FT}, model::SnowModel{FT}, Y, p) where {FT}
+    Δt_t = model.parameters.Δt / 3600 #in hours (when Δt in SnowParameters is ::FT instead of ::Period)
+
+    #Contribution from new snowfall: (uses parameterization of newsnow_temp, newsnow_density)
+    air_temp = p.drivers.T .- model.parameters.earth_param_set.T_freeze
+    snowfall = abs.(p.drivers.P_snow) .* model.parameters.Δt
+    T_newsnow = newsnow_temp.(p.drivers.T)
+    ρ_precip = newsnow_density.(air_temp) #unitless
+    ρ_newsnow =
+        compact_density.(
+            snowfall,
+            FT(0),
+            ρ_precip,
+            Δt_t,
+            T_newsnow,
+            Ref(density),
+        )
+    Δz = snowfall ./ ρ_newsnow
+
+    #Estimate resulting change from compaction effects of old snow (and the incoming snow on top):
+    W_ice = (1 .- p.snow.q_l) .* Y.snow.S
+    ρ_ice = W_ice ./ Y.snow.Z #unitless
+    sliq = p.snow.q_l .* Y.snow.S
+    Tsnow = p.snow.T .- model.parameters.earth_param_set.T_freeze
+    W_use = W_ice .+ snowfall
+    parent(ρ_ice)[parent(W_ice) .<= eps(FT)] .= FT(0.1) #any nonzero value for no-snowpack to avoid NaN and return 0.0
+    ρ_est = compact_density.(W_use, sliq, ρ_ice, Δt_t, Tsnow, Ref(density))
+    dz_compaction = (W_ice ./ ρ_est) .- Y.snow.Z
+
+    return (dz_compaction .+ Δz) ./ model.parameters.Δt
+end
+
+"""
+    clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT
+A helper function which clips the tendency of Z such that
+its behavior is consistent with that of S.
```

Comment:
"A helper function which clips the tendency of Z such that
its behavior is consistent with that of S: if all snow melts within a timestep, we clip the tendency of S so that it does not become negative, and here we also clip the tendency of Z so that depth does not become negative. Additionally, if the tendencies of Z and S are such that we would encounter Z < S (rho_snow > rho_liq), we also clip the tendency.